src/utils/vscodeUtils.ts

```diff
@@ -34,18 +35,36 @@ export async function showNewFile(data: string, fileName: string, fileExtension:
     const fullFileName: string | undefined = await getUniqueFileName(folderPath, fileName, fileExtension);
     uri = vscode.Uri.file(path.join(folderPath, fullFileName)).with({ scheme: 'untitled' });
     const textDocument = await vscode.workspace.openTextDocument(uri);
-    const editor = await vscode.window.showTextDocument(textDocument, column ? column > vscode.ViewColumn.Three ? vscode.ViewColumn.One : column : undefined, true);
+    const editor = await showTextDocument(textDocument, column);
     await writeToEditor(editor, data);
 }
 
-export async function writeToEditor(editor: vscode.TextEditor, data: string): Promise<void> {
+export async function createOrAppendToFile(data: string, fileName: string, fileExtension: string, column?: vscode.ViewColumn): Promise<void> {
```

Comment:
If you want to fix the save stuff, you could refactor [openReadonlyContent](https://github.com/microsoft/vscode-azuretools/blob/master/ui/src/openReadOnlyContent.ts) to support the `createOrAppend` scenario and use that